bundles/org.openhab.binding.bluetooth.roaming/README.md

```diff
@@ -0,0 +1,41 @@
+# Bluetooth Roaming Adapter
+
+This extension adds support for accessing Bluetooth devices from any other configured adapter via a virtual adapter.
+
+## Supported Things
+
+It defines the following bridge type:
+
+| Bridge Type ID | Description                                                                                                |
+|----------------|------------------------------------------------------------------------------------------------------------|
+| roaming        | A virtual Bluetooth adapter that interacts with Bluetooth devices through their nearest Bluetooth adapter. |
+
+## Channels
+
+Devices which use a `roaming` adapter as their bridge also gain the following channels:
+
+| Channel ID       | Item Type | Description                                                     |
+|------------------|-----------|-----------------------------------------------------------------|
+| adapter-uid      | String    | The thingUID of the adapter that is nearest to this device      |
+| adapter-location | String    | The value of the `Location` specified for the nearest adapter    |
+
+## Discovery
+
+A default roaming adapter is discovered automatically, although you can still configure it textually if you want.
+There can be only a single roaming adapter on a system.
```

Comment:
If this is due to technical constraints, it smells like some unclean design. In general, it should be allowed to create any number of things/bridges of a given type.

Reply:
Well I was worried a bit of how that particular aspect would blow over. Technically I could support more than one but since every roaming adapter monitors every other real adapter I didn't see a reason to allow supporting more than one since they would both report the same thing. So the roaming adapter limitation was more a design choice rather than a technical limitation. I can add support for it if you think that is what I should do.

I suppose down the line multiple roaming adapters could come in handy if I added support for manually choosing which real adapters a roaming adapter would monitor. Then you could have multiple roaming adapters each responsible for a separate set of real adapters.

But I don't know if that is something worth pursuing at the moment since the number of bluetooth adapters available to openhab is limited to the number that can be directly connected to the device that openhab is running on (might not be the case for bluegiga adapters if you use serial->net software).
But once ble->mqtt support comes down the line then adding that feature would very likely need to be considered.

WDYT?

Reply:
If it is not so much a technical constraint, but rather a functional/logical one, then this sounds good to me. I mainly wanted to make sure that such "logic simplifications" do not result in shortcuts in the code, which would go against the architectural guidelines (to have things independent from each other and individually referenceable).
No need to make any use of the fact that there might be more than one through features in the code right now.